### PR TITLE
feat: helm reference

### PR DESCRIPTION
**Pull request by adityabhatia**
**Description**

Changes proposed in this pull request:

- [x]  Provide a Helm-based exemplary operator
- [x] Update documentation for API reference and controller implementation

**Related issue(s)**

### CODE REVIEW

**Review by lindnerby — APPROVED** [review]

File: controllers/sample_local_helm_controller.go [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+// HandleDeletingState processed the deletion on the reconciled resource.
+// Once the deletion if processed the relevant finalizers (if applied) are removed.
+func (r *SampleHelmReconciler) HandleDeletingState(ctx context.Context, objectInstance *v1alpha1.SampleHelm) error {
+	r.Event(objectInstance, "Normal", "Deleting", "resource deleting")
+	logger := log.FromContext(ctx)
+
+	status := objectInstance.Status
+
+	resources, err := r.render(ctx, objectInstance)
+	if err != nil && controllerutil.RemoveFinalizer(objectInstance, finalizer) {
```
Comment (Tomasz-Smelcerz-SAP) [inline controllers/sample_local_helm_controller.go:157]: What if `controllerutil.RemoveFinalizer` returns false?

**Comment by Tomasz-Smelcerz-SAP** [comment]
`make install` doesn't work for me:
```
~/Documents/development/golang/src/github.com/kyma-project/template-operator/bin/controller-gen rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
~/Documents/development/golang/src/github.com/kyma-project/template-operator/bin/kustomize build config/crd | kubectl apply -f -
Error: error converting YAML to JSON: yaml: line 8: did not find expected key
error: no objects passed to apply
make: *** [Makefile:82: install] Error 1
```

**Review by Tomasz-Smelcerz-SAP — CHANGES_REQUESTED** [review]
Something is misconfigured in the kustomization.
In addition, there's no EventRecorder attached to the new `SampleHelmReconciler` which causes panics in runtime.

File: main.go [hunk 2]
```diff
@@ -103,10 +105,17 @@ func main() {
...
 		Scheme:        mgr.GetScheme(),
 		EventRecorder: mgr.GetEventRecorderFor(operatorName),
-	}).SetupWithManager(mgr, ratelimiter, flagVar.chartPath); err != nil {
+	}).SetupWithManager(mgr, ratelimiter); err != nil {
 		setupLog.Error(err, "unable to create controller", "controller", "Sample")
 		os.Exit(1)
 	}
+	if err = (&controllers.SampleHelmReconciler{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
```
Comment (Tomasz-Smelcerz-SAP) [inline main.go:114]: No `EventRecorder` here causes panics in runtime - unable to send events (r.Event == nil)

File: controllers/sample_local_helm_controller.go (hunk 1 above)
Comment (adityabhatia) [reply controllers/sample_local_helm_controller.go]: if it returns false, there will be simply no update call anymore

File: config/samples/operator.kyma-project.io_v1alpha1_samplehelm.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+apiVersion: operator.kyma-project.io.kyma-project.io/v1alpha1
```
Comment (Tomasz-Smelcerz-SAP) [inline config/samples/operator.kyma-project.io_v1alpha1_samplehelm.yaml:1]: Duplicated `kyma-project.io`

**Review by Tomasz-Smelcerz-SAP — DISMISSED** [review]

**Review by Tomasz-Smelcerz-SAP — APPROVED** [review]

**Pull request merged**